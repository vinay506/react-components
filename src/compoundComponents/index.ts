import TextWithTooltip from './TextWithToolTip/TextWithTooltip';
import TextWithIcon from './TextWithIcon/TextWithIcon';
import PaymentCard from './PaymentCard/paymentCard';
import CommentsCard from './commentCard/commentCard';

export {TextWithTooltip,TextWithIcon,PaymentCard,CommentsCard}