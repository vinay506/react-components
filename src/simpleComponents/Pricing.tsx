import React from 'react';
import TextPrimary from './TextPrimary/TextPrimary';
import TextContent from './TextContent/TextContent';
import PricePackCard from './PricePackCard/PricePackCard';
import styled from 'styled-components';
import reactIcon from '../images/react-icon.png';
import ContentOverCard from './ContentOverCard/ContentOverCard';
import ActivityCard from '../Dashboard/ActivityCard/ActivityCard';


function Pricing() {
    const clickMe = () => {
        console.log('clicked');
    }
    // <div style={{ width: '200px', height: '200px', border: '1px solid', position: 'relative' }}>
    //     <CustomBadge text={'Recommended'} position={'absolute'} top={'-10px'} bgColor={'grey'}></CustomBadge>
    // </div>

    // <h1>Pricing page</h1>
    //     <TextPrimary text='Packs to empower your preparation'></TextPrimary>
    //     <TextSecondary text='Packs & features'></TextSecondary>
    //     <div>
    //         <TextContent text='Compare between the features of packs, choose'></TextContent>
    //     </div>
    //     <TextBlur text="Current Pack"></TextBlur>
    //     <Link text='View Schedule' onLinkClick={clickMe} ></Link>
    //     <CustomButton type={BUTTON_TYPES.DENSE} text={'Get Started'} color={'blue'} bgColor={'white'} borderColor={'blue'} onButtonClick={clickMe}></CustomButton>
    const freePack = () => {
        console.log('free pack');
    }
    const secondPack = () => {
        console.log('second pack');
    }
    const recomendedPack = () => {
        console.log('recomonded pack');
    }

    const activityCardHandler = () => {
        console.log('Activity Card handler');
    }
    return (
        <PricingContainer>
            <br></br>
            <ActivityCard cardClick={activityCardHandler} logoImage={reactIcon} title="Tests" description="Take full, part or chapter tests to improve your performance.
"></ActivityCard>
            <br></br>

            <ContentOverCard>
                <div>
                    <TextPrimary text={'Packs to empower your preparation journey. '}></TextPrimary>
                    <TextPrimary text={'Cancel anytime with in 15 days.'}></TextPrimary>
                    <TextContent text={'Get Started with the pack best suited for you to prepare your upcoming events.'}></TextContent>
                </div>
            </ContentOverCard>
            <PricePackContainer>
                <PricePackCard titleText={'Free'}
                    packText={'Current pack'}
                    rupees={'0'} buttonText={'Get Started'}
                    buttonCallBack={freePack}></PricePackCard>
                <PricePackCard logo={reactIcon} titleText={'JEE Comprehensive 1-Year Pack (for Class 11)'}
                    packText={'1 year pack Valid till June 2021'} targetText={'Target Year 2020'} scheduleLink={'View Schedule'}
                    rupees={'3000'} expiryYearText={'for 1 year'} buttonText={'Get Started'}
                    buttonCallBack={secondPack}></PricePackCard>
                <PricePackCard badgeText={'Recommended'} logo={reactIcon} titleText={'JEE Comprehensive 1-Year Pack (for Class 11)'}
                    packText={'1 year pack Valid till June 2021'} targetText={'Target Year 2020'} scheduleLink={'View Schedule'}
                    rupees={'3000'} expiryYearText={'for 1 year'} buttonText={'Get Started'} buttonCallBack={recomendedPack}></PricePackCard>
            </PricePackContainer>
        </PricingContainer >
    )
}

const PricePackContainer = styled.div`
margin: 10px;
display: flex;
justify-content: center;
`
const PricingContainer = styled.div`
    text-align:center
`


export default Pricing
