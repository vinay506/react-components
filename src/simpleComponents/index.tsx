import TextSecondary from './TextSecondary/TextSecondary';
import TextPrimary from './TextPrimary/TextPrimary';
import TextContent from './TextContent/TextContent';
import TextBlur from './TextBlur/TextBlur';
import Link from './Link/Link';
import CustomBadge from './CustomBadge/CustomBadge';
import CustomButton from './CustomButton/CustomButton';
import CustomDropdown from './CustomDropdown/CustomDropdown';
import SquareCard from './SquareCard/SquareCard';
import RectangleCard from './RectangleCard/RectangleCard';
import RoundedBox from './RoundedBox/RoundedBox';
import CommentIcon from './Icons/CommentIcon/CommentIcon';
import NameWithSubText from './NameWithSubText/NameWithSubText';
import CustomTableComponent from './CustomTableComponent/CustomTableComponent';
import InfoIcon from './Icons/InfoIcon/InfoIcon';
import DisplayWithFlex from './DisplayWithFlex/DisplayWithFlex';
import Tooltip from './Tooltip/Tooltip';
import TextSecondaryRegular from './TextSecondaryRegular/TextSecondaryRegular';
import DateDetails from './DateDetails/DateDetails';

export {
    TextSecondary, TextPrimary, TextContent, TextBlur, Link, CustomBadge,
    CustomButton, CustomDropdown, SquareCard, RectangleCard, RoundedBox, CommentIcon,
    NameWithSubText,CustomTableComponent,InfoIcon,Tooltip,DisplayWithFlex,TextSecondaryRegular,
    DateDetails};
