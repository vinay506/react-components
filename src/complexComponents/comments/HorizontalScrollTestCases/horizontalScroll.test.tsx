import React from 'react';
import { shallow } from 'enzyme';
import Comments from '../commentsComponent';
import { TextPrimary, TextSecondary } from '../../simpleComponents';
import renderer from 'react-test-renderer';
import 'jest-styled-components'
import {CommentsCard} from './../../compoundComponents';

describe('<Horizontal scroll for comments container>', function () {
    let wrapper: any;
    let outerContainer: any;
    let cardsContainer:any;
    beforeEach(() => {

        wrapper = shallow(<Comments></Comments>);
        outerContainer = wrapper.find('.commentsOuterPart');
        const props = outerContainer.props();
        cardsContainer = shallow(props.children[1]);
    });

    // Snapshot test
    it('comments card snapshot', () => {
        const tree = renderer.create(<Comments ></Comments>).toJSON()
        expect(tree).toMatchSnapshot()
    })

   
    it('checks the number of cards rendered', () => {
      
        const cards = cardsContainer.find(CommentsCard);
         expect(cards).toHaveLength(4);
    });

    
})