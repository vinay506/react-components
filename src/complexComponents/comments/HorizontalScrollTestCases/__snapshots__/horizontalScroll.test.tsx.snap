// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Horizontal scroll for comments container> comments card snapshot 1`] = `
.c3 {
  font-size: 18px;
  color: black;
  font-family: 'AmazonEmber-bold';
}

.c2 {
  color: black;
  font-size: 21px;
  font-family: 'AmazonEmber-Bold';
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  line-height: 28px;
}

.c5 {
  padding: 15px 20px 0px 20px;
  margin: 2%;
  box-shadow: 0px 0px 3px;
  width: 350px;
  text-align: left;
  background: #fff;
}

.c7 {
  background-color: #bbb;
  border-radius: 50%;
  display: inline-block;
}

.c8 {
  color: #aecbfa;
  font-size: 12px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  background: #e8f0fe;
  padding-left: 100px;
  padding-top: 30px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow-x: auto;
}

.c1 {
  margin-bottom: 20px;
  margin-left: 2%;
}

<div
  className="c0 commentsOuterPart"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      id="textPrimary"
      style={
        Object {
          "fontSize": "16px",
          "marginBottom": "20px",
        }
      }
    >
      Trusted by 100,000 students in India
    </div>
    <div
      className="c3"
      id="textSecondary"
      style={
        Object {
          "fontSize": "12px",
          "fontWeight": "500",
        }
      }
    >
      Here from our happy customers
    </div>
  </div>
  <div
    className="c4"
  >
    <div
      className="c5"
    >
      <svg
        className="rmwc-icon rmwc-icon--component material-icons"
        height="30px"
        id="Capa_1"
        version="1.1"
        viewBox="0 0 508.044 508.044"
        width="30px"
        x="0px"
        y="0px"
      >
        <g>
          <g>
            <path
              d="M0.108,352.536c0,66.794,54.144,120.938,120.937,120.938c66.794,0,120.938-54.144,120.938-120.938
 s-54.144-120.937-120.938-120.937c-13.727,0-26.867,2.393-39.168,6.61C109.093,82.118,230.814-18.543,117.979,64.303
 C-7.138,156.17-0.026,348.84,0.114,352.371C0.114,352.426,0.108,352.475,0.108,352.536z"
            />
            <path
              d="M266.169,352.536c0,66.794,54.144,120.938,120.938,120.938s120.938-54.144,120.938-120.938S453.9,231.599,387.106,231.599
 c-13.728,0-26.867,2.393-39.168,6.61C375.154,82.118,496.875-18.543,384.04,64.303C258.923,156.17,266.034,348.84,266.175,352.371
 C266.175,352.426,266.169,352.475,266.169,352.536z"
            />
          </g>
        </g>
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
      </svg>
      <div
        className="c3"
        id="textSecondary"
        style={
          Object {
            "fontSize": "12px",
            "fontWeight": "500",
            "marginBottom": "20px",
            "marginTop": "15px",
            "whiteSpace": "break-spaces",
            "width": "300px",
          }
        }
      >
        JEE Ready has been helped me so much in my preparation. I practiced all the questions, which were also good and in the right difficulty! I was able to do massively well in JEE
      </div>
      <div
        className="c6"
      >
        <div
          className="c7"
          style={
            Object {
              "height": "30px",
              "marginBottom": "20px",
              "marginRight": "10px",
              "width": "30px",
            }
          }
        />
        <div
          className="c8"
        >
          <div
            className="nameText"
          >
            Asha Sharma
          </div>
          <div
            className="subText"
          >
            60 AIR
          </div>
        </div>
      </div>
    </div>
    <div
      className="c5"
    >
      <svg
        className="rmwc-icon rmwc-icon--component material-icons"
        height="30px"
        id="Capa_1"
        version="1.1"
        viewBox="0 0 508.044 508.044"
        width="30px"
        x="0px"
        y="0px"
      >
        <g>
          <g>
            <path
              d="M0.108,352.536c0,66.794,54.144,120.938,120.937,120.938c66.794,0,120.938-54.144,120.938-120.938
 s-54.144-120.937-120.938-120.937c-13.727,0-26.867,2.393-39.168,6.61C109.093,82.118,230.814-18.543,117.979,64.303
 C-7.138,156.17-0.026,348.84,0.114,352.371C0.114,352.426,0.108,352.475,0.108,352.536z"
            />
            <path
              d="M266.169,352.536c0,66.794,54.144,120.938,120.938,120.938s120.938-54.144,120.938-120.938S453.9,231.599,387.106,231.599
 c-13.728,0-26.867,2.393-39.168,6.61C375.154,82.118,496.875-18.543,384.04,64.303C258.923,156.17,266.034,348.84,266.175,352.371
 C266.175,352.426,266.169,352.475,266.169,352.536z"
            />
          </g>
        </g>
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
      </svg>
      <div
        className="c3"
        id="textSecondary"
        style={
          Object {
            "fontSize": "12px",
            "fontWeight": "500",
            "marginBottom": "20px",
            "marginTop": "15px",
            "whiteSpace": "break-spaces",
            "width": "300px",
          }
        }
      >
        JEE Ready has been helped me so much in my preparation. I practiced all the questions, which were also good and in the right difficulty! I was able to do massively well in JEE
      </div>
      <div
        className="c6"
      >
        <div
          className="c7"
          style={
            Object {
              "height": "30px",
              "marginBottom": "20px",
              "marginRight": "10px",
              "width": "30px",
            }
          }
        />
        <div
          className="c8"
        >
          <div
            className="nameText"
          >
            Asha Sharma
          </div>
          <div
            className="subText"
          >
            60 AIR
          </div>
        </div>
      </div>
    </div>
    <div
      className="c5"
    >
      <svg
        className="rmwc-icon rmwc-icon--component material-icons"
        height="30px"
        id="Capa_1"
        version="1.1"
        viewBox="0 0 508.044 508.044"
        width="30px"
        x="0px"
        y="0px"
      >
        <g>
          <g>
            <path
              d="M0.108,352.536c0,66.794,54.144,120.938,120.937,120.938c66.794,0,120.938-54.144,120.938-120.938
 s-54.144-120.937-120.938-120.937c-13.727,0-26.867,2.393-39.168,6.61C109.093,82.118,230.814-18.543,117.979,64.303
 C-7.138,156.17-0.026,348.84,0.114,352.371C0.114,352.426,0.108,352.475,0.108,352.536z"
            />
            <path
              d="M266.169,352.536c0,66.794,54.144,120.938,120.938,120.938s120.938-54.144,120.938-120.938S453.9,231.599,387.106,231.599
 c-13.728,0-26.867,2.393-39.168,6.61C375.154,82.118,496.875-18.543,384.04,64.303C258.923,156.17,266.034,348.84,266.175,352.371
 C266.175,352.426,266.169,352.475,266.169,352.536z"
            />
          </g>
        </g>
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
      </svg>
      <div
        className="c3"
        id="textSecondary"
        style={
          Object {
            "fontSize": "12px",
            "fontWeight": "500",
            "marginBottom": "20px",
            "marginTop": "15px",
            "whiteSpace": "break-spaces",
            "width": "300px",
          }
        }
      >
        JEE Ready has been helped me so much in my preparation. I practiced all the questions, which were also good and in the right difficulty! I was able to do massively well in JEE
      </div>
      <div
        className="c6"
      >
        <div
          className="c7"
          style={
            Object {
              "height": "30px",
              "marginBottom": "20px",
              "marginRight": "10px",
              "width": "30px",
            }
          }
        />
        <div
          className="c8"
        >
          <div
            className="nameText"
          >
            Asha Sharma
          </div>
          <div
            className="subText"
          >
            60 AIR
          </div>
        </div>
      </div>
    </div>
    <div
      className="c5"
    >
      <svg
        className="rmwc-icon rmwc-icon--component material-icons"
        height="30px"
        id="Capa_1"
        version="1.1"
        viewBox="0 0 508.044 508.044"
        width="30px"
        x="0px"
        y="0px"
      >
        <g>
          <g>
            <path
              d="M0.108,352.536c0,66.794,54.144,120.938,120.937,120.938c66.794,0,120.938-54.144,120.938-120.938
 s-54.144-120.937-120.938-120.937c-13.727,0-26.867,2.393-39.168,6.61C109.093,82.118,230.814-18.543,117.979,64.303
 C-7.138,156.17-0.026,348.84,0.114,352.371C0.114,352.426,0.108,352.475,0.108,352.536z"
            />
            <path
              d="M266.169,352.536c0,66.794,54.144,120.938,120.938,120.938s120.938-54.144,120.938-120.938S453.9,231.599,387.106,231.599
 c-13.728,0-26.867,2.393-39.168,6.61C375.154,82.118,496.875-18.543,384.04,64.303C258.923,156.17,266.034,348.84,266.175,352.371
 C266.175,352.426,266.169,352.475,266.169,352.536z"
            />
          </g>
        </g>
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
      </svg>
      <div
        className="c3"
        id="textSecondary"
        style={
          Object {
            "fontSize": "12px",
            "fontWeight": "500",
            "marginBottom": "20px",
            "marginTop": "15px",
            "whiteSpace": "break-spaces",
            "width": "300px",
          }
        }
      >
        JEE Ready has been helped me so much in my preparation. I practiced all the questions, which were also good and in the right difficulty! I was able to do massively well in JEE
      </div>
      <div
        className="c6"
      >
        <div
          className="c7"
          style={
            Object {
              "height": "30px",
              "marginBottom": "20px",
              "marginRight": "10px",
              "width": "30px",
            }
          }
        />
        <div
          className="c8"
        >
          <div
            className="nameText"
          >
            Asha Sharma
          </div>
          <div
            className="subText"
          >
            60 AIR
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
